modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -143,20 +147,84 @@ GrainTracker::finalize()
 
   expandHalos();
 
-  FeatureFloodCount::communicateAndMerge();
+  // Build up the grain map on the root processor
+  communicateAndMerge();
+
+  /**
+   * Track Grains
+   */
+  Moose::perf_log.push("trackGrains()", "GrainTracker");
+  std::vector<unsigned int> new_grain_indices;
+
+  // Track grains (only on the root processor)
+  if (_is_master)
+    trackGrains(new_grain_indices);
+
+  if (!new_grain_indices.empty())
+  {
+    // Communicate new grain indices with the remaining processors
+    auto new_grains_count = new_grain_indices.size();
+    _communicator.broadcast(new_grains_count);
+
+    new_grain_indices.resize(new_grains_count);
+    _communicator.broadcast(new_grain_indices);
+
+    // Trigger the callback on all processors
```

Comment:
Hmm, what does that look like? I'm pretty certain that @roystgnr or @jwpeterson will accept PRs that fix these things up. I'd certainly like to continue making the `parallel.h` tools easier to use and more efficient. I've got https://github.com/libMesh/libmesh/pull/1067 up there now that's waiting to be merged and it's sending separate broadcasts (or scatters) for sizes instead of probing.